Sources/StreamChat/WebSocketClient/EventMiddlewares/ChannelReadUpdaterMiddleware.swift

```diff
@@ -6,20 +6,22 @@ import Foundation
 
 /// A middleware which updates a channel's read events as websocket events arrive.
 struct ChannelReadUpdaterMiddleware: EventMiddleware {
-    func handle(event: Event, session: DatabaseSession) -> Event? {
+    func handle(event: Event, session: DatabaseSession, notificationCenter: EventNotificationCenter) -> Event? {
```

Comment:
Ideally, we should not break the `EventMiddleware` interface and only add the notification center to this middleware, which is the only one who needs it. This, or provide a protocol or closure to fetch the `newMessageIds`.

This way we only need to change this middleware, and not all of them. This PR will be much smaller as a result of this. 

```suggestion
    weak var notificationCenter: EventNotificationCenter?
    init(notificationCenter: EventNotificationCenter) {
         self.notificationCenter = notificationCenter
    }
    
    func handle(event: Event, session: DatabaseSession) -> Event? {
```